core/src/smartcontracts/isi/block.rs

```diff
@@ -9,11 +13,44 @@ impl ValidQuery for FindAllBlocks {
     fn execute(&self, wsv: &WorldStateView) -> Result<Self::Output, query::Error> {
         let mut blocks: Vec<Box<BlockValue>> = wsv
             .blocks()
-            .map(|blk| blk.clone())
+            .map(|block| block.clone())
             .map(VersionedCommittedBlock::into_value)
             .map(Box::new)
             .collect();
-        blocks.reverse();
+        blocks.reverse(); // Sorted by height desc.
```

Comment:
you can reverse by iterating
```rust
            .rev() // Sorted by height desc.
            .collect(); 
```

Reply:
good catch! `ChainIterator` implements`DoubleEndedIterator` so it's possible.